Yoast/wordpress-seo
# Add feature flag for greek

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Adds feature flag for Greek Support.

## Relevant technical choices:

* This PR should be tested with the[ accompanying Premium PR](https://github.com/Yoast/wordpress-seo-premium/pull/3450).

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* In PHPStorm/your preferred IDE, navigate to `packages/yoastseo/src/languageProcessing/languages/_default` and copy the contents
* Inside the `languages` folder, create an `el` folder and paste the contents from `_default` (don't commit any changes)
* Build the free plugin
* Enable the `GREEK_SUPPORT` feature by adding `define( 'YOAST_SEO_GREEK_SUPPORT', true );` in `basic-wordpress-config.php` file
* Set your site to Greek `Ελληνικά`
* Create/open a post
* In the console, enter `window.yoast.Researcher` and confirm that it returns as shown below (the Researcher file should start with `el`):
<img width="482" alt="Screenshot 2021-09-20 at 17 28 32" src="https://user-images.githubusercontent.com/48715883/134184644-8c0260f4-da74-48f2-9660-5cb51fe5ade4.png">

* Run this command `composer require yoast/wordpress-seo:dev-LINGO-1042-add-feature-flag-for-Greek@dev` in `wordpress-seo-premium`
* Build the premium plugin
* Reload your page
* In your console, confirm that the morphology request returns with `404`:
<img width="468" alt="Screenshot 2021-09-20 at 17 27 28" src="https://user-images.githubusercontent.com/48715883/134184688-80660f36-4a61-45e5-8ac1-aec8f4b2580f.png">

* Disable the `GREEK_SUPPORT` by adding this  `define( 'YOAST_SEO_GREEK_SUPPORT', false );` in `basic-wordpress-config.php` file
* Reload your page
* Confirm that you get the default Researcher by entering `window.yoast.Researcher` in the console:

<img width="455" alt="Screenshot 2021-09-21 at 15 59 15" src="https://user-images.githubusercontent.com/48715883/134184747-7f848d9c-a1be-401b-8118-190e59c96edd.png">


* Confirm that there is no morphology request
<img width="728" alt="Screenshot 2021-06-17 at 17 11 19" src="https://user-images.githubusercontent.com/48715883/122424791-49c74080-cf8f-11eb-8880-728b64259b8d.png">



### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Install and activate Free plugin
* Enable the `GREEK_SUPPORT` feature by adding `define( 'YOAST_SEO_GREEK_SUPPORT', true );` in `basic-wordpress-config.php` file
* Set your site to Greek `Ελληνικά`
* Create/open a post
* In the console, enter `window.yoast.Researcher` and confirm that it returns as shown below (the Researcher file should start with `el`):
<img width="482" alt="Screenshot 2021-09-20 at 17 28 32" src="https://user-images.githubusercontent.com/48715883/134184644-8c0260f4-da74-48f2-9660-5cb51fe5ade4.png">

* Install and activate Premium plugin
* Reload your page
* In your console, confirm that the morphology request returns with `404`:
<img width="468" alt="Screenshot 2021-09-20 at 17 27 28" src="https://user-images.githubusercontent.com/48715883/134184688-80660f36-4a61-45e5-8ac1-aec8f4b2580f.png">

* Disable the `GREEK_SUPPORT` by adding this  `define( 'YOAST_SEO_GREEK_SUPPORT', false );` in `basic-wordpress-config.php` file
* Reload your page
* Confirm that you get the default Researcher by entering `window.yoast.Researcher` in the console:
<img width="455" alt="Screenshot 2021-09-21 at 15 59 15" src="https://user-images.githubusercontent.com/48715883/134184747-7f848d9c-a1be-401b-8118-190e59c96edd.png">


* Confirm that there is no morphology request
<img width="728" alt="Screenshot 2021-06-17 at 17 11 19" src="https://user-images.githubusercontent.com/48715883/122424791-49c74080-cf8f-11eb-8880-728b64259b8d.png">


## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-1042
